### PR TITLE
open file

### LINKED ISSUE #88: std_misc/file
1. [open.md](https://github.com/nebula-labs/rust-by-example-vn/blob/main/src/std_misc/file/open.md)
2. [read_lines.md](https://github.com/nebula-labs/rust-by-example-vn/blob/main/src/std_misc/file/read_lines.md)

### PR DESCRIPTION
**Pull request by phongnguyen2012**
Closes: #88 
https://github.com/nebula-labs/rust-by-example-vn/blob/main/src/std_misc/file/open.md
https://github.com/nebula-labs/rust-by-example-vn/blob/main/src/std_misc/file/read_lines.md

### CODE REVIEW

File: src/std_misc/file/open.md [hunk 1]
```diff
@@ -1 +1,45 @@
-# open
+# `open`
+
+Hàm `open` có thể được sử dụng để mở một tập tin ở chế độ chỉ được đọc.
+
+Môt `File` sở hữu một tài nguyên, đó là mô tả tập tin và sẽ tự động đóng khi không còn (sử dụng) `drop`.
```
Comment (HongThaiPham) [inline src/std_misc/file/open.md]:
```suggestion
Một `File` sở hữu một tài nguyên, đó là mô tả tập tin và sẽ tự động đóng khi không còn (sử dụng) `drop`.
```

File: src/std_misc/file/open.md [hunk 2]
```diff
@@ -1 +1,45 @@
...
+Dưới đây là kết quả thành công mong đợi:
+
+```shell
+$ echo "Hello World!" > hello.txt
+$ rustc open.rs && ./open
+hello.txt contains:
+Hello World!
+```
+
+(Bạn được khuyến khích thử nghiệm ví dụ trước đó trong các điều kiện thất bại
```
Comment (HongThaiPham) [inline src/std_misc/file/open.md]:
```suggestion
(Khuyến khích bạn nên thử nghiệm ví dụ trên trong các điều kiện có thể xảy ra lỗi 
```

File: src/std_misc/file/read_lines.md [hunk 3]
```diff
@@ -1 +1,87 @@
...
+chúng ta cũng có thể thực hiện một phép ánh xạ trực tiếp và thu thập các kết quả, 
+tạo ra một biểu thức ngắn gọn và trôi chảy hơn
+
+```rust,norun
+use std::fs::read_to_string;
+
+fn read_lines(filename: &str) -> Vec<String> {
+    read_to_string(filename) 
+        .unwrap()  // panic khi có lỗi đọc tệp có thể xảy ra.
+        .lines()  // chia chuỗi thành một terator of string slices
```
Comment (HongThaiPham) [inline src/std_misc/file/read_lines.md]:
```suggestion
        .lines()  // chia chuỗi thành một bộ lặp của các dòng 
```

File: src/std_misc/file/read_lines.md [hunk 4]
```diff
@@ -1 +1,87 @@
...
+}
+```
+
+Lưu ý rằng trong cả hai ví dụ trên, chúng ta phải chuyển đổi tham chiếu `&str` 
+được trả về từ `lines()` thành kiểu được sở hữu `String`, sử dụng `.to_string()` và 
+`String::from` tương ứng.
+
+## Một cách tiếp cận hiệu quả hơn
+
+Ở đây chúng ta truyền sở hữu của `File` đã mở vào một ` struct. ` BufReader.
```
Comment (HongThaiPham) [inline src/std_misc/file/read_lines.md]:
```suggestion
Ở đây chúng ta truyền sở hữu của `File` đã mở vào một struct `BufReader`. 
```

File: src/std_misc/file/read_lines.md [hunk 5]
```diff
@@ -1 +1,87 @@
...
+
+Chạy chương trình này đơn giản chỉ in các dòng một cách riêng lẻ.
+```shell
+$ echo -e "127.0.0.1\n192.168.0.1\n" > hosts.txt
+$ rustc read_lines.rs && ./read_lines
+127.0.0.1
+192.168.0.1
+```
+
+(Lưu ý rằng vì `File::open` mong đợi một AsRef<Path> chung như đối số, chúng ta xác định
```
Comment (HongThaiPham) [inline src/std_misc/file/read_lines.md]:
```suggestion
(Lưu ý rằng vì `File::open` mong đợi một generic AsRef<Path> như đối số, chúng ta xác định 
```

File: src/std_misc/file/read_lines.md [hunk 6]
```diff
@@ -1 +1,87 @@
...
+Chạy chương trình này đơn giản chỉ in các dòng một cách riêng lẻ.
+```shell
+$ echo -e "127.0.0.1\n192.168.0.1\n" > hosts.txt
+$ rustc read_lines.rs && ./read_lines
+127.0.0.1
+192.168.0.1
+```
+
+(Lưu ý rằng vì `File::open` mong đợi một AsRef<Path> chung như đối số, chúng ta xác định 
+phương thức `read_lines()` chung của mình với cùng một ràng buộc chung, sử dụng từ khóa `where`.)
```
Comment (HongThaiPham) [inline src/std_misc/file/read_lines.md]:
```suggestion
phương thức generic `read_lines()` với cùng một ràng buộc generic, sử dụng từ khóa `where`.)
```

**Review by HongThaiPham — APPROVED** [review]

File: src/std_misc/file/open.md [hunk 7]
```diff
@@ -1 +1,45 @@
-# open
+# `open`
+
+Hàm `open` có thể được sử dụng để mở một tập tin ở chế độ chỉ được đọc.
+
+Một `File` sở hữu một tài nguyên, đó là mô tả tập tin và sẽ tự động đóng khi không còn (sử dụng) `drop`.
```
Comment (duonghb53) [inline src/std_misc/file/open.md]:
```suggestion
Một `File` sở hữu một tài nguyên, đó là mô tả tập tin và đảm bảo file được đóng lại khi tài nguyên đó được giải phóng `drop`.
```

File: src/std_misc/file/read_lines.md (hunk 6 above)
Comment (phongnguyen2012) [reply src/std_misc/file/read_lines.md]: Done

**Comment by duonghb53** [comment]
LGTM

**Pull request merged**